src/app/core/interceptors/jwt.interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from "@angular/core";
+import {
+  HttpInterceptor,
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpErrorResponse
+} from "@angular/common/http";
+import { Observable } from "rxjs";
+import { tap } from "rxjs/operators";
+import { Router } from '@angular/router';
+@Injectable()
+export class JwtInterceptor implements HttpInterceptor {
+    constructor(private router: Router) {}
+
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return next.handle(req).pipe(tap((event: HttpEvent<any>) => {
```

Comment:
Why `tap`?

Reply:
because in documentation it's used "do" method. but it's deprecated and the equivalent is tap method in rxjs

Reply:
If you have to catch only errors use catchError instead of tap